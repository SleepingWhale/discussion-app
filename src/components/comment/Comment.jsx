import React from 'react';
import PropTypes from 'prop-types';
import styles from './Comment.module.css';
import { UserPick } from '../common';

export function Comment({ content, author }) {
  return (
    <div className={styles.container}>
      <UserPick url={author.pick} title={author.name} />
      <div className={styles.comment}>{content}</div>
    </div>
  );
}

Comment.propTypes = {
  content: PropTypes.string,
  author: PropTypes.shape({
    name: PropTypes.string,
    nickname: PropTypes.string,
    pick: PropTypes.string
  }).isRequired
};

Comment.defaultProps = {
  content: ''
};
