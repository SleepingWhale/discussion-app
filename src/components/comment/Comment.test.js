import React from 'react';
import { shallow } from 'enzyme';
import { Comment } from './Comment';

const author = {
  name: 'Agent Smith',
  nickname: '@smithy',
  pick: 'http://placekitten.com/32/32?image=2'
};
const content = 'content';

describe('Comment', () => {
  it('should render correctly', () => {
    const component = shallow(<Comment author={author} content={content} />);
    expect(component).toMatchSnapshot();
  });
});
