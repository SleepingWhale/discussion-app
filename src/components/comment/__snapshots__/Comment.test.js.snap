// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should render correctly 1`] = `
<div
  className="container"
>
  <UserPick
    title="Agent Smith"
    url="http://placekitten.com/32/32?image=2"
  />
  <div
    className="comment"
  >
    content
  </div>
</div>
`;
